vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -1108,11 +1108,11 @@ WIFIReturnCode_t WIFI_NetworkDelete( uint16_t usIndex )
 			{
 				xRet = nvs_commit( xNvsHandle );
 			}
-		}
 
-		if( xRet == ESP_OK )
-		{
-			xWiFiRet = eWiFiSuccess;
+			if( xRet == ESP_OK )
```

Comment:
Perhaps, replacing[ entry to this _success_ block](https://github.com/aws/amazon-freertos/pull/2909/files#diff-b2e9f40dcf41678b337ff2ac1fa8ae4c06535ea4296c33fe419c319b4e90c245R1092)
```c
if( xRet == ESP_OK && usIndex < xRegistry.usNumNetworks )
```
with the following instead:
```c
if( xRet == ESP_OK && (0 < usIndex && usIndex < xRegistry.usNumNetworks ))
```
The latter spares runtime of block [1094, 1110] and may be more straightforward failure path when `usIndex == 0`